pymc/distributions/multivariate.py

```diff
@@ -1415,7 +1422,71 @@ def LKJCholeskyCov(name, eta, n, sd_dist, compute_corr=False, store_in_trace=Tru
         return chol, corr, stds
 
 
-class LKJCorr(Continuous):
+class LKJCorrRV(RandomVariable):
+    name = "lkjcorr"
+    ndim_supp = 1
+    ndims_params = [0, 0]
+    dtype = "floatX"
+    _print_name = ("LKJCorrRV", "\\operatorname{LKJCorrRV}")
+
+    def make_node(self, rng, size, dtype, n, eta):
+        n = at.as_tensor_variable(n)
+        if not n.ndim == 0:
+            raise ValueError("n must be a scalar (ndim=0).")
+
+        eta = at.as_tensor_variable(eta)
+        if not eta.ndim == 0:
+            raise ValueError("eta must be a scalar (ndim=0).")
+
+        return super().make_node(rng, size, dtype, n, eta)
+
+    def _shape_from_params(self, dist_params, **kwargs):
+        n = dist_params[0]
+        dist_shape = ((n * (n - 1)) // 2,)
+        return dist_shape
+
+    @classmethod
+    def rng_fn(cls, rng, n, eta, size):
+
+        # We flatten the size to make operations easier, and then rebuild it
+        if size is None:
+            orig_size = None
+            size = 1
+        else:
+            orig_size = size
+            size = np.prod(size)
+
+        # original implementation in R see:
+        # https://github.com/rmcelreath/rethinking/blob/master/R/distributions.r
+        beta = eta - 1.0 + n / 2.0
+        r12 = 2.0 * stats.beta.rvs(a=beta, b=beta, size=size, random_state=rng) - 1.0
+        P = np.full((size, n, n), np.eye(n))
+        P[..., 0, 1] = r12
+        P[..., 1, 1] = np.sqrt(1.0 - r12 ** 2)
+        for mp1 in range(2, n):
+            beta -= 0.5
+            y = stats.beta.rvs(a=mp1 / 2.0, b=beta, size=size, random_state=rng)
+            z = stats.norm.rvs(loc=0, scale=1, size=(size, mp1), random_state=rng)
+            z = z / np.sqrt(np.einsum("ij,ij->i", z, z))[..., np.newaxis]
```

Comment:
@tomicapretto, continuing from https://github.com/pymc-devs/pymc/pull/4784#issuecomment-1019983326 here is where I think the not yet refactored `LKJCholeskyCov` random method was wrong. My understanding is that it was trying to generalize the pre-existing code in `LJKCorr` (removed in this PR) to allow for more flexible sizes, but in doing so altered the meaning of this einsum. The still "buggy" code there for reference is this:  

https://github.com/pymc-devs/pymc/blob/0b5f8d134f7842faf5820f4c40ae737b05b81174/pymc/distributions/multivariate.py#L1183

CC @lucianopaz, I think you wrote this code originally, do you have a chance to ping in?

Reply:
I remember that PR being centered around making a mixture distribution of MvNormals work, and to be able to sample from their prior. The flexible `size` stuff came from over there. I hope that I did not mess up the einsum back then, but I honestly don't remember why I had written `"j"` instead of `"i"`, and I don't remember the algorithm of the rng at all.

Reply:
When I used this new more flexible logic for the LKJCorrRV, the random test failed, and that seems to be the stricter test for the rng that we have. 

Then I started debugging line by line, and the einsum index was what changed the results between the old logic in `LKJCorrRV` and the more flexible one in `LKJCholeskyRV`.